proxy/src/outbound.rs

```diff
@@ -38,33 +41,52 @@ impl<B> Outbound<B> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Destination {
+    LocalSvc(FullyQualifiedAuthority),
+    External(SocketAddr),
```

Comment:
Why don't we need `External` to be `External(Authority)` or `External(IpAddr, Authority)`?

In particular, in HTTP/1 we're generally not supposed to send requests for different `Host:`s on the same connection because not all HTTP/1 servers properly support that. This makes me think that both the authority and the IP address must be part of the key.

(OTOH, if we were already mixing requests for different hosts on the same outbound HTTP/1 connection then we can address this in a separate issue later.)

Regardless of which way we do it, let's add a comment explaining why we are doing it the way we're doing it, and not the other way.

Reply:
The addition of the `Host` header in HTTP/1.1 is actually specifically for that purpose, to allow a server to handle requests for various authorities.

Additionally, it's possible at many stages already to find that a `Host` is found at the same address. I don't believe this PR changes anything.

Reply:
1. How do we know the *server* is HTTP/1.1 and not HTTP/1.0? We would have had to have previously received a HTTP/1.1 response from the server to know that; we can't know it from looking at the request.

2. Although I mostly agree with you w.r.t. what was intended in HTTP/1.1, many HTTP/1.1 servers don't properly respect `Host:` beyond the first request, which is why browsers don't (IIRC) send requests for different `Host`s on the same connection. This is why, in particular, browsers don't do connection coalescing for HTTP/1.1, but only for HTTP/2.

Regardless, you don't have to answer those questions here. I just want to point out that it is important to not regress in this respect. As long as you're sure that the behavior isn't changed from before, we can deal with it in another issue.